interfaces/builtin/raw_usb.go

```diff
@@ -37,6 +37,9 @@ const rawusbConnectedPlugAppArmor = `
 # Allow access to all ttyUSB devices too
 /dev/tty{USB,ACM}[0-9]* rwk,
 
+# Allow raw access to USB printers (i.e. for receipt printers in POS systems).
+/dev/usb/lp[0-9]* rwk,
```

Comment:
This looks fine based on our prior discussion. I do not have access to a USB printer though and I wonder if we need to tweak the udev backend. We currently have:
```
var rawusbConnectedPlugUDev = []string{
        `SUBSYSTEM=="usb"`,
        `SUBSYSTEM=="tty", ENV{ID_BUS}=="usb"`,
}
```

Is `SUBSYSTEM=="usb"` enough for this to work?

I *think* it might be since looking at 50-udev-default.rules I see:
```
KERNEL=="parport[0-9]*", GROUP="lp"
SUBSYSTEM=="printer", KERNEL=="lp*", GROUP="lp"
SUBSYSTEM=="ppdev", GROUP="lp"
KERNEL=="lp[0-9]*", GROUP="lp"
KERNEL=="irlpt[0-9]*", GROUP="lp"
SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ENV{ID_USB_INTERFACES}=="*:0701??:*", GROUP="lp"
```

while there is a "printer" SUBSYSTEM, this last line is what I believe is for USB printers. Our `SUBSYSTEM==usb` is a superset of this line, so it should be there. Can someone with a USB printer comment on if a snap that plugs 'raw-usb' ends up having the /dev/usb/lp0 (or similar) tagged? Eg, after hotplugging the printer:

```
# in one terminal
$ sudo snap install something-with-raw-usb
$ snap run --shell something-with-raw-usb.cmd

# in another terminal
$ udevadm info /dev/usb/lp0  # adjust as necessary
...
E: TAGS=:snap_something-with-raw-usb_cmd:
```
(ie, we want to see the snap in TAGS)

Reply:
the one usb printer i own is a HP which forcefully turns off the creation of the USB devices (like all HP ones do to avoid clashes, they talk directly to the printer via the usb bus through their own userspace driver)  so i can not test this here either, i'd suggest we land it as is and in case there are any issues later when the customer tries to use it we/I can adjust.

Reply:
That's fine. Thanks for the extra info.